infrastructure/modules/cdn-frontdoor/variables.tf

```diff
@@ -0,0 +1,130 @@
+variable "cdn_frontdoor_firewall_policy_rg_name" {
+  description = "Resource Group name containing Front Door WAF policies. You may optionally override this per firewall policy in var.security_policies"
+  type        = string
+}
+
+variable "cdn_frontdoor_profile_id" {
+  description = "ID of the Front Door profile to associate endpoints and origin groups with"
+  type        = string
+}
+
+variable "custom_domains" {
+  description = "Map of Front Door Custom Domain configurations"
+  type = map(object({
+    dns_zone_name = string
+    host_name     = string
+
+    tls = optional(object({
+      # https://learn.microsoft.com/en-us/azure/frontdoor/standard-premium/how-to-configure-https-custom-domain?tabs=powershell#register-azure-front-door
+      certificate_type        = optional(string, "ManagedCertificate") # Use of apex domain as a hostname requires "CustomerCertificate"
+      cdn_frontdoor_secret_id = optional(string, null)
+    }), {})
+
+    zone_rg_name = optional(string, null)
+  }))
+}
+
+variable "endpoint" {
+  description = "Front Door Endpoint configuration"
+  type = object({
+    enabled = optional(bool, true)
+  })
+  default = {}
+}
+
+variable "environment" {
```

Comment:
As discussed this could be left to user to set the environment in the name. So this could be removed